include/dynd/kernels/arithmetic.hpp

```diff
@@ -68,7 +68,49 @@ namespace nd {
                                          &detail::inline_logical_xor<Src0TypeID, Src1TypeID>::f> {
   };
 
-#undef DYND_DeclBinopKernel
+#define DYND_DEF_BINARY_OP_KERNEL_ZEROCHECK_INT(OP, NAME)                                                              \
+  namespace detail {                                                                                                   \
+    template <type_id_t Src0TypeID, type_id_t Src1TypeID,                                                              \
+              type_id_t Src0Base = base_id_of<Src0TypeID>::value,                                                      \
+              type_id_t Src1Base = base_id_of<Src1TypeID>::value,                                                      \
+              bool check = ((Src0Base == bool_kind_id) || (Src0Base == int_kind_id) || (Src0Base == uint_kind_id)) &&  \
+                           ((Src1Base == bool_kind_id) || (Src1Base == int_kind_id) || (Src1Base == uint_kind_id))>    \
+    struct inline_##NAME;                                                                                              \
+                                                                                                                       \
+    template <type_id_t Src0TypeID, type_id_t Src1TypeID, type_id_t Src0Base, type_id_t Src1Base>                      \
```

Comment:
I did things in that order to avoid adding an extra layer of indirection in the templates before passing the kernel in to `make_all` at https://github.com/libdynd/libdynd/blob/master/include/dynd/func/arithmetic.hpp#L61. Is there another interface there that I should be using or are you asking for another set of templates here?